### PR TITLE
namespace JWTTypes as individual

### PR DESCRIPTION
**Pull request by ekoeryanto**
Related to #96 and #98 

cc: @Ethan-Arrowood 

feel free to close

#### Checklist

- [ ] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message and code follows [Code of conduct](https://github.com/fastify/fastify/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Comment by Ethan-Arrowood** [comment]
Can you add/update tests? This looks good!

**Comment by ekoeryanto** [comment]
hello, any news abot this?

File: jwt.test-d.ts [hunk 1]
```diff
@@ -26,7 +26,7 @@ app.register(fastifyJwt, {
         authorizationTokenInvalid: (err) => `${err.message}`,
         authorizationTokenUntrusted: 'Token untrusted'
     },
-    trusted: () => true,
+    trusted: () => false || '' || Buffer.from('foo'),
```
Comment (mcollina) [inline jwt.test-d.ts:29]: Is this the only test that is needed? Is this code only to check if `JWTTypes` are outside of the fastify namespace? Can you please add a test for those?

Comment (ekoeryanto) [reply jwt.test-d.ts:29]: > Is this the only test that is needed?

`boolean | Promise<boolean> | JWTTypes.SignPayloadType | Promise<JWTTypes.SignPayloadType>`
it missed the promise types, I'm going to add the promise types test

> Is this code only to check if JWTTypes are outside of the fastify namespace?

No, it doesn't check for `JWTTypes`, I don't think we need to check it.
It only fixes miss-reference to (unknown) JWTTypes from outside the namespace.

**Comment by mcollina** [comment]
thanks!

**Comment by ekoeryanto** [comment]
I did hope it released in the `next` tag, but it was one hour missed. 😢

**Comment by ekoeryanto** [comment]
I still have build issue with version 2.1.2 😱

**Pull request merged**